src/main/scala/analysis/InterprocSteensgaardAnalysis.scala

```diff
@@ -0,0 +1,395 @@
+package analysis
+
+import analysis.solvers.{Cons, Term, UnionFindSolver, Var}
+import ir.*
+import util.Logger
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+
+/** Wrapper for variables so we can have Steensgaard-specific equals method indirectly */
+sealed trait VariableWrapper {
+  val variable: Variable
+}
+
+/** Wrapper for variables so we can have Steensgaard-specific equals method indirectly */
+case class RegisterVariableWrapper(variable: Variable) extends VariableWrapper {
+    override def equals(obj: Any): Boolean = {
+        obj match {
+        case RegisterVariableWrapper(other) =>
+            variable == other && variable.ssa_id.intersect(other.ssa_id).nonEmpty
+        case _ =>
+            false
+        }
+    }
+
+    override def hashCode(): Int = {
+        variable.hashCode()
+    }
+}
```

Comment:
This is a very messy way to handle SSA. If we are using this wrapper, why did mutable state need to be added to the Variable trait? Don't override the hashcode for a case class.